### PR TITLE
Capture on order done & refund/release on delivery return

### PR DESCRIPTION
**Pull request by maf-embraceit**

### CODE REVIEW

**Review by shahbaz-mehar — COMMENTED** [review]
Changes have been reviewed.

File: src/Subscriber/OrderStateChangeSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+                    $response = $this->paymentService->captureReservation($order, $order->getSalesChannelId(), $remainingAmount);
+                    $responseAsXml = new SimpleXMLElement($response->getBody()->getContents());
+
+                    if ((string)$responseAsXml->Body?->Result === "Success") {
+                        $this->orderTransactionStateHandler->paid(
+                            $order->getTransactions()->first()->getId(),
+                            $context
+                        );
+                    }
```
Comment (emicha) [inline src/Subscriber/OrderStateChangeSubscriber.php:65]: What if it is not successful? Shouldn't we log the error?

File: src/Subscriber/OrderStateChangeSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                    $order->getSalesChannelId()
+                );
+                $responseAsXml = new SimpleXMLElement($response->getBody()->getContents());
+
+                if ((string)$responseAsXml->Body?->Result === "Success") {
+                    $this->orderTransactionStateHandler->refund(
+                        $order->getTransactions()->first()->getId(),
+                        $context
+                    );
+                }
```
Comment (emicha) [inline src/Subscriber/OrderStateChangeSubscriber.php:112]: Same as above

File: src/Subscriber/OrderStateChangeSubscriber.php [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+                    $order->getSalesChannelId()
+                );
+                $responseAsXml = new SimpleXMLElement($response->getBody()->getContents());
+
+                if ((string)$responseAsXml->Body?->Result === "Success") {
+                    $this->orderTransactionStateHandler->cancel(
+                        $order->getTransactions()->first()->getId(),
+                        $context
+                    );
+                }
```
Comment (emicha) [inline src/Subscriber/OrderStateChangeSubscriber.php:127]: Same as above

File: src/Subscriber/OrderStateChangeSubscriber.php (hunk 1 above)
Comment (BushraAsif) [reply src/Subscriber/OrderStateChangeSubscriber.php]: @emicha , I’ve added the log.

File: src/Subscriber/OrderStateChangeSubscriber.php (hunk 2 above)
Comment (BushraAsif) [reply src/Subscriber/OrderStateChangeSubscriber.php]: Logs added!

File: src/Subscriber/OrderStateChangeSubscriber.php (hunk 3 above)
Comment (BushraAsif) [reply src/Subscriber/OrderStateChangeSubscriber.php]: Logs added!

**Review by emicha — APPROVED** [review]

**Pull request merged**